apollo-compiler/src/main/kotlin/com/apollostack/compiler/GraphqlCompiler.kt

```diff
@@ -34,12 +34,12 @@ open class GraphqlCompiler {
 
   private fun String.formatPackageName(): String {
     val parts = split(File.separatorChar)
-    val srcFolderIndex = parts.indexOfFirst { it == "src" }
-    val graphqlFolderIndex = parts.indexOfFirst { it == "graphql" }
-    if (graphqlFolderIndex - srcFolderIndex != 2) {
-      throw IllegalArgumentException("Files must be organized like src/main/graphql/...")
-    }
 
-    return parts.subList(graphqlFolderIndex + 1, parts.size).dropLast(1).joinToString(".")
+    for (i in 2..parts.size) {
+      if (parts[i - 2] == "src" && parts[i] == "graphql") {
```

Comment:
It is, but before we're finding the first occurrence of 'src' and 'graphql' and doing the distance check. Now, it loops through the whole path string.

The one we had before breaks when the first occurrence for the case I mentioned above: when there are two `src` directories in the string.